static/js/theme-map/VerticalFullPageMapOrchestrator.js

```diff
@@ -478,14 +494,42 @@ class VerticalFullPageMapOrchestrator extends ANSWERS.Component {
     }
   }
 
+  /**
+   * Update the map wrapper's "top" and "height" css properties so that it is not covered up
+   * by the no results block in mobile. 
+   *
+   * @param {ResultsContext} resultsContext 
+   */
+  updateMapDimensionsForNoResults(resultsContext) {
+    if (!this.isInMobileMapView) {
+      return;
+    }
+    const noResults = this._noResultsContainer;
+    const map = document.querySelector(this._mapContainerSelector);
+    if (!noResults || !map) {
+      return;
+    }
+    if (resultsContext === ResultsContext.NO_RESULTS) {
+      const noResultsHeight = noResults.getBoundingClientRect().height;
+      map.style.top =
```

Comment:
I'm a bit worried about having CSS directly applied from the JS. It makes it hard to pinpoint where styling is coming from and haves us worried about two different places for a page's styling when making changes. However I don't see another way because

(1) flex-grow cannot be used as the `mapWrapper` is not sibling to the results header and results footer
(2) the map should be fixed to contain the height on mobile phones

No change request, just voicing the concern

Reply:
Would it be possible to toggle a css class as usual, and include some new variable, say `--yxt-no-results-height`, which we update from the javascript when we toggle the no results css class?

Reply:
This is what i'm thinking: https://css-tricks.com/updating-a-css-variable-with-javascript/

Reply:
oh wow I like that idea a lot, I was trying out a different way of setting the height from what I have in the PR and was running into occasional desync issues (with the update method running before some other random part of the page was done) but I think setting a variable might fix it